msnc/model.py

```diff
@@ -79,32 +78,29 @@ def run_training(
         Keyword Arguments:
             epoch_num {int} -- number of epochs (default: {100})
             checkpoint_interval {int} -- it creates checkpoints at {checkpoint_interval} (default: {10})  # NOQA
-            save_best_model {} -- save the best model if True (default: {True})
         """
         self._output_dir_path = pathlib.Path(output_dir)
         best_dev_accuracy, best_epoch = -float('inf'), 0
 
         batches = training_set.split(self.batch_size)
         for epoch in range(1, epoch_num + 1):
+            self.epoch = epoch
             self.train()
-            self._train(batches, epoch)
+            self._train(batches)
 
-            if not self._ischeckpoint(epoch, checkpoint_interval):
+            if not self._ischeckpoint(checkpoint_interval):
                 continue
-            model_file_name = '{:04d}.model'.format(epoch)
-            self._save(model_file_name)
 
             dev_accuracy = self.run_evaluation(development_set)
-            self._write_log(dev_accuracy, epoch)
+            self._write_log(dev_accuracy, self.epoch)
             if dev_accuracy > best_dev_accuracy:
-                best_dev_accuracy, best_epoch = dev_accuracy, epoch
-                self._write_log(dev_accuracy, epoch, '[new best]')
-                if save_best_model:
-                    self._save('best.model')
+                best_dev_accuracy, best_epoch = dev_accuracy, self.epoch
+                self._save('best.model')
```

Comment:
I think this implementation does not guarantee to save the best model.
This is because `self._save` only fires when `epoch % checkpoint_interval == 0`.
(For example,  `checkout_interval=5` and model achieves the best at `epoch=12`.)

Is it OK? (If not, I think we can avoid this by putting `self._ischeckpoint` just before `model._save`, L:97)

Reply:
This comment is completely wrong... :bow:

Reply:
Thank you for the comment. I marge this PR 🙏